server/sonar-web/src/main/js/apps/custom-metrics/components/App.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { MetricProps } from './Form';
+import Header from './Header';
+import List from './List';
+import {
+  getMetricDomains,
+  getMetricTypes,
+  getAllMetrics,
+  deleteMetric,
+  updateMetric,
+  createMetric
+} from '../../../api/metrics';
+import { Metric } from '../../../app/types';
+import ListFooter from '../../../components/controls/ListFooter';
+import { translate } from '../../../helpers/l10n';
+
+interface Props {}
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: An empty interface is equivalent to `{}`. ([typescript:S4023](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4023))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fcustom-metrics&issues=AWF1nUAY0orRW2HVTuy-&open=AWF1nUAY0orRW2HVTuy-)

[ISSUE_KEY]: AWF1nUAY0orRW2HVTuy-